src/sentry/api/bases/project.py

```diff
@@ -111,25 +112,42 @@ class ProjectEndpoint(Endpoint):
     def convert_args(
         self,
         request: Request,
-        organization_slug: str,
-        project_slug: str,
+        organization_slug: str | int,
+        project_slug: str | int,
         *args,
         **kwargs,
     ):
         try:
-            project = (
-                Project.objects.filter(organization__slug=organization_slug, slug=project_slug)
-                .select_related("organization")
-                .prefetch_related("teams")
-                .get()
-            )
+            if options.get("api.id-or-slug-enabled") and str(organization_slug).isnumeric():
```

Comment:
I just realized do we need this logic of checking if the slug is numeric. I think the lookup handles this for us

Reply:
you are right, copy pasting has made my brain 😵

Reply:
it's also in the except case too

Reply:
fixed!